nengo/dists.py

```diff
@@ -309,11 +328,52 @@ class SubvectorLength(SqrtBeta):
     --------
     SqrtBeta
     """
+
     def __init__(self, dimensions, subdimensions=1):
         super(SubvectorLength, self).__init__(
             dimensions - subdimensions, subdimensions)
 
 
+class CosineSimilarity(SubvectorLength):
+    """Distribution of dot products between random unit vectors.
+
+    This is equivalent to the distribution of a single coefficient in a unit
+    vector (i.e. a single dimension of `UniformHypersphere(surface=True)`),
+    which can be obtained from `SubvectorLength` by symmetry.
+
+    This can be used to calculate an intercept `c = invcdf(1 - p)` such that
```

Comment:
*facepalm*

On Wed, Dec 9, 2015 at 5:35 PM, Aaron Voelker <notifications@github.com>
wrote:

> In nengo/dists.py
> <https://github.com/nengo/nengo/pull/768#discussion_r47163356>:
>
> >      def __init__(self, dimensions, subdimensions=1):
> >          super(SubvectorLength, self).__init__(
> >              dimensions - subdimensions, subdimensions)
> >
> >
> > +class CosineSimilarity(SubvectorLength):
> > +    """Distribution of dot products between random unit vectors.
> > +
> > +    This is equivalent to the distribution of a single coefficient in a unit
> > +    vector (i.e. a single dimension of `UniformHypersphere(surface=True)`),
> > +    which can be obtained from `SubvectorLength` by symmetry.
> > +
> > +    This can be used to calculate an intercept `c = invcdf(1 - p)` such that
>
> It means "Inverse Cumulative Distribution Function". The cdf part is
> standard terminology. This method name has already been used elsewhere a
> couple places in the file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nengo/nengo/pull/768/files#r47163356>.
>